src/linear_algebra/prod.lean

```diff
@@ -539,6 +539,14 @@ lemma prod_eq_top_iff {p₁ : submodule R M} {p₂ : submodule R M₂} :
   p₁.prod p₂ = ⊤ ↔ p₁ = ⊤ ∧ p₂ = ⊤ :=
 by simp only [eq_top_iff, le_prod_iff, ← (gc_map_comap _).le_iff_le, map_top, range_fst, range_snd]
 
+lemma mem_map_fst_iff {p : submodule R (M × M₂)} {x : M} :
+  x ∈ p.map (linear_map.fst R M M₂) ↔ ∃ (y : M₂), (x,y) ∈ p :=
+by simp
```

Comment:
Can you add this lemma for `set` too? The proof of this lemma should then be defeq to that one.

Reply:
I would guess it should go next to `_root_.prod.range_fst`

Reply:
I've added the lemma for `set` and it does not prove the lemma for `submodule`.

Reply:
Does it not? I find that quite surprising